File: 2358-number-of-ways-to-split-array/2358-number-of-ways-to-split-array.java
class Solution {
    public int waysToSplitArray(int[] nums) {
        List<Long> prefixSum = new ArrayList<>();
        long totalSum = 0;
        for(int i:nums){
            totalSum += (long)i;
            prefixSum.add(totalSum);
        }
        int n = nums.length;
        int res = 0;
        for(int i = 0;i < n - 1;i++){
            long leftSum = prefixSum.get(i);
            if(leftSum >= (totalSum - leftSum)) res++;
        }
        return res;
    }
}

File: 2358-number-of-ways-to-split-array/2358-number-of-ways-to-split-array.py
class Solution:
    def waysToSplitArray(self, nums: List[int]) -> int:
        total_sum = 0
        for i in nums:
            total_sum += i
        
        res = 0
        left_sum = 0
        for i in range(0,len(nums) - 1):
            left_sum += nums[i]
            if(left_sum >= (total_sum - left_sum)):
                res += 1
            
        return res